coalib/core/Bear.py

```diff
@@ -130,6 +132,21 @@ class Bear:
     CAN_FIX = set()
     ASCIINEMA_URL = ''
     BEAR_DEPS = set()
+    """
+        Attributes for a new bear.
+
+        :param section:
+            The section object where bear settings are contained.
+        :param file_dict:
+            The file-dictionary containing a mapping of filenames to the
+            according file contents.
+        :raises RuntimeError:
+            Raised when bear requirements are not fulfilled.
+    """
+    section = attr.ib(hash=False)
+    file_dict = attr.ib(hash=False)
+    _dependency_results = attr.ib(init=False, default=defaultdict(list),
+                                  hash=False)
```

Comment:
> we can obtain all attributes as a dict or json

this is in fact already possible by using `vars(instance)` or `dir(instance)` ;)

> We can attach hooks to each attribute which will execute after an attribute is initialized 

so that's what we would do inside the `__init__` anyway, right?

> This PR just adds the attrs, not It's usecases :)

Yeah I understood that intention already from @sils :) But the bear is in fact working, so I don't really see to change something in terms of "layout".

Don't want to sound too strict or so, I just want to settle things with `attrs`, as it came accross me from some people, telling me like "oh man super cool stuff", but I can't see that for `coalib.core.Bear` :/